theano/compile/debugmode.py

```diff
@@ -1537,6 +1558,23 @@ def _check_preallocated_output(node, thunk, prealloc_modes, def_val,
                         specific_hint=r.type.value_validity_msg(
                             storage_map[r][0]))
 
+                if check_data_ptr:
+                    # If the preallocated output has been reused, check
+                    # that it has the same data pointer
+                    var = storage_map[r][0]
+                    if init_output_vars[i] is var:
+                        if hasattr(var, 'data'):
+                            data = var.data
+                        elif hasattr(var, 'gpudata'):
+                            data = var.gpudata
+                        else:
+                            data = None
+
+                        if data != init_output_data[i]:
+                            _logger.warning("The node %s reuses its "
```

Comment:
We want to make this an error in another commit in this PR while we change the op that behave like this in the same PR. This will be prohibited from now on. It won't be just something that isn't optimal or acceptable but not optimal.

It is a good idea to help find node that do this to just warn about it. Just in case there is many of them in the same graph.